src/extensions/markdown/Image/imageUrlPaste/index.ts

```diff
@@ -0,0 +1,55 @@
+import {Plugin} from 'prosemirror-state';
+
+import {ExtensionAuto} from '../../../../core';
+import {DataTransferType} from '../../../behavior/Clipboard/utils';
+import {imageType} from '../ImageSpecs';
+
+export type ImageUrlPasteOptions = {
+    /**
+     * The function, used to determine if the pasted text is the image url and should be inserted as an image
+     */
+    shouldInsertPastedUrlAsImage?: (text: string) => boolean;
+    /**
+     * The function, used to get the image alternative text from the pasted image url
+     */
+    getImageNameFromPastedUrl?: (imageUrl: string) => string;
```

Comment:
I suggest combining these two functions into one that returns either an object or `null`, the object may contains the title

for example:
```
parseInsertedUrlAsImage?: (text: string) => {imageUrl: string, title?: string} | null;
```